GPy/util/block_matrices.py

```diff
@@ -2,36 +2,39 @@
 # Licensed under the BSD 3-clause license (see LICENSE.txt)
 import numpy as np
 
+
 def get_blocks_3d(A, blocksizes, pagesizes=None):
     """
     Given a 3d matrix, make a block matrix, where the first and second dimensions are blocked according
     to blocksizes, and the pages are blocked using pagesizes
     """
-    assert (A.shape[0]==A.shape[1]) and len(A.shape)==3, "can't blockify this non-square matrix, may need to use 2d version"
+    assert (A.shape[0] == A.shape[1]) and len(
```

Comment:
This one's a bit ugly ... but it also makes me wonder if assert should be used in these cases. 

Would the right thing be to raise a ValueError?

Reply:
Yeah, the right thing would be `ValueError`. In general, I find the following block extracted from https://wiki.python.org/moin/UsingAssertionsEffectively to be a good guide.

"""
Assertions should *not* be used to test for failure cases that can occur because of bad user input or operating system/environment failures, such as a file not being found. Instead, you should raise an exception, or print an error message, or whatever is appropriate. One important reason why assertions should only be used for self-tests of the program is that assertions can be disabled at compile time.
"""

Following the notion that `assert` shouldn't address bad user input - in this case, the shape of `A` - it should indeed instead be a `ValueError`.

That being said, this suggestion is outside the scope of this PR - I'm just making linting and styling changes here. I'll file an issue, one sec...

Reply:
https://github.com/SheffieldML/GPy/issues/922